planner/core/runtime_filter_generator.go

```diff
@@ -0,0 +1,242 @@
+		generator.childIdxForParentPhysicalPlan = 0
+		generator.GenerateRuntimeFilter(physicalPlan.tablePlan)
+		if physicalPlan.StoreType == kv.TiFlash {
+			physicalPlan.TablePlans = flattenPushDownPlan(physicalPlan.tablePlan)
+		}
+	}
+
+	for i, children := range plan.Children() {
+		generator.parentPhysicalPlan = plan
+		generator.childIdxForParentPhysicalPlan = i
+		generator.GenerateRuntimeFilter(children)
+	}
+}
+
+func (generator *RuntimeFilterGenerator) generateRuntimeFilterInterval(hashJoinPlan *PhysicalHashJoin) {
+	// precondition: the storage type of hash join must be TiFlash
+	if hashJoinPlan.storeTp != kv.TiFlash {
+		logutil.BgLogger().Warn("RF only support TiFlash compute engine while storage type of hash join node is not TiFlash",
+			zap.Int("PhysicalHashJoinId", hashJoinPlan.id),
+			zap.String("StoreTP", hashJoinPlan.storeTp.Name()))
+		return
+	}
+	// check hash join pattern
+	if !generator.matchRFJoinType(hashJoinPlan) {
+		return
+	}
+	// check eq predicate pattern
+	for _, eqPredicate := range hashJoinPlan.EqualConditions {
+		if generator.matchEQPredicate(eqPredicate, hashJoinPlan.RightIsBuildSide()) {
+			// construct runtime filter
+			newRFList, targetColumnUniqueID := NewRuntimeFilter(generator.rfIDGenerator, eqPredicate, hashJoinPlan)
+			// update generator rf list
+			rfList := generator.columnUniqueIDToRF[targetColumnUniqueID]
+			if rfList == nil {
+				generator.columnUniqueIDToRF[targetColumnUniqueID] = newRFList
+			} else {
+				generator.columnUniqueIDToRF[targetColumnUniqueID] = append(generator.columnUniqueIDToRF[targetColumnUniqueID], newRFList...)
+			}
+		}
+	}
+}
+
+func (generator *RuntimeFilterGenerator) assignRuntimeFilter(physicalTableScan *PhysicalTableScan) {
+	// match rf for current scan node
+	cacheBuildNodeIDToRFMode := map[int]RuntimeFilterMode{}
+	var currentRFList []*RuntimeFilter
+	for _, scanOutputColumn := range physicalTableScan.schema.Columns {
+		currentColumnRFList := generator.columnUniqueIDToRF[scanOutputColumn.UniqueID]
+		for _, runtimeFilter := range currentColumnRFList {
+			// compute rf mode
+			var rfMode RuntimeFilterMode
+			if cacheBuildNodeIDToRFMode[runtimeFilter.buildNode.id] != 0 {
+				rfMode = cacheBuildNodeIDToRFMode[runtimeFilter.buildNode.id]
+			} else {
+				rfMode = generator.calculateRFMode(runtimeFilter.buildNode, physicalTableScan)
+				cacheBuildNodeIDToRFMode[runtimeFilter.buildNode.id] = rfMode
+			}
+			// todo support global RF
+			switch rfMode {
+			case variable.RFGlobal:
+				logutil.BgLogger().Debug("Now we don't support global RF. Remove it",
+					zap.Int("BuildNodeId", runtimeFilter.buildNode.id),
+					zap.Int("TargetNodeId", physicalTableScan.id))
+				continue
+			}
+			runtimeFilter.rfMode = rfMode
+
+			// assign rf to current node
+			runtimeFilter.assign(physicalTableScan, scanOutputColumn)
+			currentRFList = append(currentRFList, runtimeFilter)
+		}
+	}
+
+	if len(currentRFList) == 0 {
+		return
+	}
+
```

Comment:
what does the usage of `currentRFList`？

Reply:
Oh, this is a future code for physical selection. If the currentRFList is not empty, we need to consider whether adding a selection upper than table scan node